### PR TITLE
chore(react): Rename componentWillMount to componentDidMount

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Will see if tests all pass here

### CODE REVIEW

File: src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx [hunk 1]
```diff
@@ -128,10 +128,10 @@ export default function getOnboardingTasks(organization: Organization): Onboardi
       title: t('Set up an alerts service'),
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx]: oops, will remove this in a sec, wanna see if travis passes.

File: src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ class PluginComponentBase extends React.Component {
     };
   }
 
-  componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
   }
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: Does it make sense to initialize API client in the constructor? There's no benefit for it to be placed in any component lifecycle methods.

File: src/sentry/static/sentry/app/components/onboardingWizard/todoList.jsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ class TodoList extends React.Component {
       return task;
     });
 
+    // eslint-disable-next-line react/no-did-mount-set-state
     this.setState({tasks});
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/components/onboardingWizard/todoList.jsx]: 👀

File: src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: Yeah probably makes more sense

**Comment by EvanPurkhiser** [comment]
Billy has something similar to this already 🤷‍♂

**Comment by billyvg** [comment]
I think it'd be worth applying this PR before the UNSAFE codemod

**Comment by EvanPurkhiser** [comment]
Going to supersede this since we ran the codemod

**Pull request closed without merging**